### ISSUE TITLE
negative value for "net.ipv4.tcp_rmem" max

### ISSUE THREAD

**Issue by gecon** [oracle-samples/bpftune#36]
Tuning of "net.ipv4.tcp_rmem" max is showing some negative values (an overflow?).

```
Jul 14 04:14:49 hostname bpftune[1237369]: Scenario 'need to increase TCP buffer size(s)' occurred for tunable 'net.ipv4.tcp_rmem' in global ns. Need to increase buffer size(s) to maximize throughput
Jul 14 04:14:49 hostname bpftune[1237369]: Due to need to increase max buffer size to maximize throughput change net.ipv4.tcp_rmem(min default max) from (4096 87380 1734723440) -> (4096 87380 -2126562996)
Jul 14 04:14:49 hostname bpftune[1237369]: Scenario 'need to increase TCP buffer size(s)' occurred for tunable 'net.ipv4.tcp_rmem' in global ns. Need to increase buffer size(s) to maximize throughput
Jul 14 04:14:49 hostname bpftune[1237369]: Due to need to increase max buffer size to maximize throughput change net.ipv4.tcp_rmem(min default max) from (4096 87380 1734723440) -> (4096 87380 -2126562996)
Jul 14 04:14:49 hostname bpftune[1237369]: Scenario 'need to increase TCP buffer size(s)' occurred for tunable 'net.ipv4.tcp_rmem' in global ns. Need to increase buffer size(s) to maximize throughput
Jul 14 04:14:49 hostname bpftune[1237369]: Due to need to increase max buffer size to maximize throughput change net.ipv4.tcp_rmem(min default max) from (4096 87380 1734723440) -> (4096 87380 -2126562996)
Jul 14 04:14:49 hostname bpftune[1237369]: Scenario 'need to increase TCP buffer size(s)' occurred for tunable 'net.ipv4.tcp_rmem' in global ns. Need to increase buffer size(s) to maximize throughput
Jul 14 04:14:49 hostname bpftune[1237369]: Due to need to increase max buffer size to maximize throughput change net.ipv4.tcp_rmem(min default max) from (4096 87380 1734723440) -> (4096 87380 -2126562996)
Jul 14 04:14:49 hostname bpftune[1237369]: Scenario 'need to increase TCP buffer size(s)' occurred for tunable 'net.ipv4.tcp_rmem' in global ns. Need to increase buffer size(s) to maximize throughput
Jul 14 04:14:49 hostname bpftune[1237369]: Due to need to increase max buffer size to maximize throughput change net.ipv4.tcp_rmem(min default max) from (4096 87380 1734723440) -> (4096 87380 -2126562996)
Jul 14 04:14:49 hostname bpftune[1237369]: Scenario 'need to increase TCP buffer size(s)' occurred for tunable 'net.ipv4.tcp_rmem' in global ns. Need to increase buffer size(s) to maximize throughput
Jul 14 04:14:49 hostname bpftune[1237369]: Due to need to increase max buffer size to maximize throughput change net.ipv4.tcp_rmem(min default max) from (4096 87380 1734723440) -> (4096 87380 -2126562996)
Jul 14 04:14:49 hostname bpftune[1237369]: Scenario 'need to increase TCP buffer size(s)' occurred for tunable 'net.ipv4.tcp_rmem' in global ns. Need to increase buffer size(s) to maximize throughput
Jul 14 04:14:49 hostname bpftune[1237369]: Due to need to increase max buffer size to maximize throughput change net.ipv4.tcp_rmem(min default max) from (4096 87380 1734723440) -> (4096 87380 -2126562996)
Jul 14 04:14:49 hostname bpftune[1237369]: Scenario 'need to increase TCP buffer size(s)' occurred for tunable 'net.ipv4.tcp_rmem' in global ns. Need to increase buffer size(s) to maximize throughput
Jul 14 04:14:49 hostname bpftune[1237369]: Due to need to increase max buffer size to maximize throughput change net.ipv4.tcp_rmem(min default max) from (4096 87380 1734723440) -> (4096 87380 -2126562996)
Jul 14 04:14:49 hostname bpftune[1237369]: Scenario 'need to increase TCP buffer size(s)' occurred for tunable 'net.ipv4.tcp_rmem' in global ns. Need to increase buffer size(s) to maximize throughput
Jul 14 04:14:49 hostname bpftune[1237369]: Due to need to increase max buffer size to maximize throughput change net.ipv4.tcp_rmem(min default max) from (4096 87380 1734723440) -> (4096 87380 -2126562996)
Jul 14 04:14:50 hostname bpftune[1237369]: Scenario 'need to increase TCP buffer size(s)' occurred for tunable 'net.ipv4.tcp_rmem' in global ns. Need to increase buffer size(s) to maximize throughput
Jul 14 04:14:50 hostname bpftune[1237369]: Due to need to increase max buffer size to maximize throughput change net.ipv4.tcp_rmem(min default max) from (4096 87380 1734723440) -> (4096 87380 -2126562996)
Jul 14 04:14:50 hostname bpftune[1237369]: Scenario 'need to increase TCP buffer size(s)' occurred for tunable 'net.ipv4.tcp_rmem' in global ns. Need to increase buffer size(s) to maximize throughput
Jul 14 04:14:50 hostname bpftune[1237369]: Due to need to increase max buffer size to maximize throughput change net.ipv4.tcp_rmem(min default max) from (4096 87380 1734723440) -> (4096 87380 -2126562996)
Jul 14 04:14:50 hostname bpftune[1237369]: Scenario 'need to increase TCP buffer size(s)' occurred for tunable 'net.ipv4.tcp_rmem' in global ns. Need to increase buffer size(s) to maximize throughput
Jul 14 04:14:50 hostname bpftune[1237369]: Due to need to increase max buffer size to maximize throughput change net.ipv4.tcp_rmem(min default max) from (4096 87380 1734723440) -> (4096 87380 -2126562996)
Jul 14 04:14:50 hostname bpftune[1237369]: Scenario 'need to increase TCP buffer size(s)' occurred for tunable 'net.ipv4.tcp_rmem' in global ns. Need to increase buffer size(s) to maximize throughput
Jul 14 04:14:50 hostname bpftune[1237369]: Due to need to increase max buffer size to maximize throughput change net.ipv4.tcp_rmem(min default max) from (4096 87380 1734723440) -> (4096 87380 -2126562996)
Jul 14 04:14:50 hostname bpftune[1237369]: Scenario 'need to increase TCP buffer size(s)' occurred for tunable 'net.ipv4.tcp_rmem' in global ns. Need to increase buffer size(s) to maximize throughput
Jul 14 04:14:50 hostname bpftune[1237369]: Due to need to increase max buffer size to maximize throughput change net.ipv4.tcp_rmem(min default max) from (4096 87380 1734723440) -> (4096 87380 -2126562996)
Jul 14 04:14:50 hostname bpftune[1237369]: Scenario 'need to increase TCP buffer size(s)' occurred for tunable 'net.ipv4.tcp_rmem' in global ns. Need to increase buffer size(s) to maximize throughput
Jul 14 04:14:50 hostname bpftune[1237369]: Due to need to increase max buffer size to maximize throughput change net.ipv4.tcp_rmem(min default max) from (4096 87380 1734723440) -> (4096 87380 -2126562996)
Jul 14 04:14:50 hostname bpftune[1237369]: Scenario 'need to increase TCP buffer size(s)' occurred for tunable 'net.ipv4.tcp_rmem' in global ns. Need to increase buffer size(s) to maximize throughput
Jul 14 04:14:50 hostname bpftune[1237369]: Due to need to increase max buffer size to maximize throughput change net.ipv4.tcp_rmem(min default max) from (4096 87380 1734723440) -> (4096 87380 -2126562996)
Jul 14 04:14:50 hostname bpftune[1237369]: Scenario 'need to increase TCP buffer size(s)' occurred for tunable 'net.ipv4.tcp_rmem' in global ns. Need to increase buffer size(s) to maximize throughput
Jul 14 04:14:50 hostname bpftune[1237369]: Due to need to increase max buffer size to maximize throughput change net.ipv4.tcp_rmem(min default max) from (4096 87380 1734723440) -> (4096 87380 -2126562996)
Jul 14 04:14:50 hostname bpftune[1237369]: Scenario 'need to increase TCP buffer size(s)' occurred for tunable 'net.ipv4.tcp_rmem' in global ns. Need to increase buffer size(s) to maximize throughput
Jul 14 04:14:50 hostname bpftune[1237369]: Due to need to increase max buffer size to maximize throughput change net.ipv4.tcp_rmem(min default max) from (4096 87380 1734723440) -> (4096 87380 -2126562996)
Jul 14 04:14:50 hostname bpftune[1237369]: Scenario 'need to increase TCP buffer size(s)' occurred for tunable 'net.ipv4.tcp_rmem' in global ns. Need to increase buffer size(s) to maximize throughput
Jul 14 04:14:50 hostname bpftune[1237369]: Due to need to increase max buffer size to maximize throughput change net.ipv4.tcp_rmem(min default max) from (4096 87380 1734723440) -> (4096 87380 -2126562996)
Jul 14 04:14:50 hostname bpftune[1237369]: Scenario 'need to increase TCP buffer size(s)' occurred for tunable 'net.ipv4.tcp_rmem' in global ns. Need to increase buffer size(s) to maximize throughput
```

OS/Kernel info:

```
[root@hostname~]# cat /etc/oracle-release
Oracle Linux Server release 8.8
[root@hostname ~]# uname -a
Linux hostname 5.4.17-2136.318.7.1.el8uek.x86_64 #2 SMP Thu Apr 13 17:33:39 PDT 2023 x86_64 x86_64 x86_64 GNU/Linux

**Comment by alan-maguire** [comment]
thanks for the report! should be fixed by #38

**Comment by gecon** [comment]
That was fast! Thanks! 👍
